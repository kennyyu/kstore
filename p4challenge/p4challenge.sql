CREATE DATABASE IF NOT EXISTS p4challenge;
USE p4challenge;

DROP TABLE IF EXISTS t;
DROP TABLE IF EXISTS u;

CREATE TABLE IF NOT EXISTS t (
a INTEGER,
b INTEGER,
c INTEGER,
d INTEGER);

CREATE TABLE IF NOT EXISTS u (
a INTEGER,
b INTEGER,
c INTEGER,
d INTEGER);

LOAD DATA LOCAL INFILE 't.csv' INTO TABLE t
FIELDS TERMINATED BY ',' ENCLOSED BY '"'
LINES TERMINATED BY '\n'
IGNORE 1 LINES
(a,b,c,d);

LOAD DATA LOCAL INFILE 'u.csv' INTO TABLE t
FIELDS TERMINATED BY ',' ENCLOSED BY '"'
LINES TERMINATED BY '\n'
IGNORE 1 LINES
(a,b,c,d);

SELECT avg(t.d), avg(u.c)
FROM t,u
WHERE t.a = u.a
        AND t.b = 1000
        AND u.d BETWEEN 9000 AND 9000000;

-- (575250,498194)

INSERT INTO t(a,b,c,d) VALUES (412234,123231,54323,764543),
(1312312,5321234,3453241,14232342),
(39485394,123124,23429,1293),
(234231,1000,23425,21234),
(30293,123901,23491,12303);

SELECT avg(t.d), avg(u.c)
FROM t,u
WHERE t.a = u.a
        AND t.b = 1000
        AND u.d BETWEEN 9000 AND 9000000;

-- (536077,507767)

DELETE FROM t WHERE d < 100000;

SELECT avg(t.d), avg(u.c)
FROM t,u
WHERE t.a = u.a
        AND t.b = 1000
        AND u.d BETWEEN 9000 AND 9000000;

-- (618527,499023)

UPDATE u SET c = 10000 WHERE b > 500000;

SELECT avg(t.d), avg(u.c)
FROM t,u
WHERE t.a = u.a
        AND t.b = 1000
        AND u.d BETWEEN 9000 AND 9000000;

-- (618527,238469)

DELETE FROM u WHERE a BETWEEN 100000 AND 200000;

SELECT avg(t.d), avg(u.c)
FROM t,u
WHERE t.a = u.a
        AND t.b = 1000
        AND u.d BETWEEN 9000 AND 9000000;

-- (618527,238469)

UPDATE t SET a = 54321 WHERE c BETWEEN 10000 AND 200000;

SELECT avg(t.d), avg(u.c)
FROM t,u
WHERE t.a = u.a
        AND t.b = 1000
        AND u.d BETWEEN 9000 AND 9000000;

-- (593609,244244)
