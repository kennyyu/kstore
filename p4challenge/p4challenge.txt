tb_select=select(tb,1000)
ta_fetch=fetch(ta,tb_select)
ud_select=select(ud,9000,9000000)
ua_fetch=fetch(ua,ud_select)
t_id,u_id=hashjoin(ta_fetch,ua_fetch)
td_fetch=fetch(td,t_id)
td_avg=avg(td_fetch)
uc_fetch=fetch(uc,u_id)
uc_avg=avg(uc_fetch)
tuple(td_avg,uc_avg)
insert(ta,412234,tb,123231,tc,54323,td,764543)
insert(ta,1312312,tb,5321234,tc,3453241,td,14232342)
insert(ta,39485394,tb,123124,tc,23429,td,1293)
insert(ta,234231,tb,1000,tc,23425,td,21234)
insert(ta,30293,tb,123901,tc,23491,td,12303)
tb_select=select(tb,1000)
ta_fetch=fetch(ta,tb_select)
ud_select=select(ud,9000,9000000)
ua_fetch=fetch(ua,ud_select)
t_id,u_id=hashjoin(ta_fetch,ua_fetch)
td_fetch=fetch(td,t_id)
td_avg=avg(td_fetch)
uc_fetch=fetch(uc,u_id)
uc_avg=avg(uc_fetch)
tuple(td_avg,uc_avg)
t_delete=select(td,0,99999)
delete(t_delete,ta,tb,tc,td)
tb_select=select(tb,1000)
ta_fetch=fetch(ta,tb_select)
ud_select=select(ud,9000,9000000)
ua_fetch=fetch(ua,ud_select)
t_id,u_id=hashjoin(ta_fetch,ua_fetch)
td_fetch=fetch(td,t_id)
td_avg=avg(td_fetch)
uc_fetch=fetch(uc,u_id)
uc_avg=avg(uc_fetch)
tuple(td_avg,uc_avg)
u_update=select(ub,500001,2147483647)
update(u_update,uc,10000)
tb_select=select(tb,1000)
ta_fetch=fetch(ta,tb_select)
ud_select=select(ud,9000,9000000)
ua_fetch=fetch(ua,ud_select)
t_id,u_id=hashjoin(ta_fetch,ua_fetch)
td_fetch=fetch(td,t_id)
td_avg=avg(td_fetch)
uc_fetch=fetch(uc,u_id)
uc_avg=avg(uc_fetch)
tuple(td_avg,uc_avg)
u_delete=select(ua,100000,200000)
delete(u_delete,ua,ub,uc,ud)
tb_select=select(tb,1000)
ta_fetch=fetch(ta,tb_select)
ud_select=select(ud,9000,9000000)
ua_fetch=fetch(ua,ud_select)
t_id,u_id=hashjoin(ta_fetch,ua_fetch)
td_fetch=fetch(td,t_id)
td_avg=avg(td_fetch)
uc_fetch=fetch(uc,u_id)
uc_avg=avg(uc_fetch)
tuple(td_avg,uc_avg)
t_update=select(tc,10000,200000)
update(t_update,ta,54321)
tb_select=select(tb,1000)
ta_fetch=fetch(ta,tb_select)
ud_select=select(ud,9000,9000000)
ua_fetch=fetch(ua,ud_select)
t_id,u_id=hashjoin(ta_fetch,ua_fetch)
td_fetch=fetch(td,t_id)
td_avg=avg(td_fetch)
uc_fetch=fetch(uc,u_id)
uc_avg=avg(uc_fetch)
tuple(td_avg,uc_avg)
