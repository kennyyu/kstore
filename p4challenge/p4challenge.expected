(575250,498194)
(536077,507767)
(618527,499023)
(618527,238469)
(618527,238469)
(593609,244244)
